telemetry/telemetry_record.py

```diff
@@ -96,9 +156,356 @@ def failed_url(url, output):
+            '/home/jamshed/page_load_time/telemetry/data/results.db'
+    results_data = {}
+    try:
+        results_data = pickle.load(open(results_path, 'rb'))
+    except IOError:
+        raise IOError('Could not read from {0}'.format(results_path))
+
+    wpr_path = os.path.join(os.getcwd(), 'data/wpr_source')
+    wpr_files = filter(lambda x: '.wpr' in x, os.listdir(wpr_path))
+    for wpr_file in wpr_files:
+        curr_wpr = cPickle.load(open(os.path.join(wpr_path, wpr_file), 'rb'))
+        curr_har_dict = {'log':
+                            {'pages':
+                                [{'id': '',
+                                 'title': '',
+                                 'pageTimings': {
+                                        'onContentLoad': None,
+                                        'onLoad': None
+                                        }
+                                 }],
+                             'entries': []
+                            }
+                        }
+        wpr_host = None
+        for key, value_lst in zip(curr_wpr.keys(), curr_wpr.values()):
+            matches = [full_url for full_url in results_data.keys() if \
+                    key.host in full_url]
+            if matches:
+                assert len(matches) == 1, ('Found more than 1 match for'
+                        ' {0}'.format(matches))
+                agg_key = matches[0]
+                # Found a match!
+                curr_har_dict['log']['pages'][0]['id'] = key.host
+                curr_har_dict['log']['pages'][0]['title'] = key.host  # Set both
+                if '999999' in wpr_file:
+                    wpr_host = b64encode(key.host) + '.pc'
+                    # It's a modified wpr file
+                    curr_har_dict['log']['pages'][0]['pageTimings']['onLoad'] \
+                            = results_data[agg_key]['modified_cold_time']
+                else:
+                    wpr_host = b64encode(key.host)
+                    # It's an original wpr file
+                    curr_har_dict['log']['pages'][0]['pageTimings']['onLoad'] \
+                            = results_data[agg_key]['cold_time']
+            else:
+                # This website is loading a third party object
+                # Still include this in the har file
+                pass
+            # Add to each element to entries
+            # Request data
+            method = key.command
+            element_url = key.host + key.full_path
+            # Response data
+            status = value_lst.status
+            headerSize = 0  # Need to find this in value_lst
+            bodySize = 0  # Need to find this in value_lst
+            # Create each element's header list
+            tmp_header_lst = []
+            for name, value in value_lst.headers:
+                if 'content-length' in name:
+                    bodySize = int(value)
+                headerSize += len(name) + len(value)  # This should be verified
```

Comment:
total size of all headers, in bytes. So, it should be close enough to count the number of characters in all header keys and values